### PR TITLE
feat(cli): Add unordered xcodebuild command support

### PR DESCRIPTION
**Pull request by yusufozgul**
Improve xcodebuild command argument parsing flexibility

> This PR enhances the `tuist xcodebuild` command to support flexible argument ordering, allowing xcodebuild actions to be placed anywhere in the command line arguments, not just as the first argument after `tuist xcodebuild`.

### Problem

Previously, the `tuist xcodebuild` command required the action (`build`, `test`, `archive`, etc.) to be the first argument due to ArgumentParser's subcommand structure:

```bash
# ✅ This worked
tuist xcodebuild build -workspace MyApp.xcworkspace -scheme MyApp

# ❌ This failed
tuist xcodebuild -workspace MyApp.xcworkspace build -scheme MyApp
```

This limitation was particularly problematic when using tools like Fastlane, which often append the action at the end of the command.

### Solution

Implemented a fallback mechanism that:

1. **Preserves existing subcommand behavior**: Standard usage continues to work exactly as before
2. **Adds flexible argument parsing**: When no subcommand is matched initially, the main command parses all arguments to find the action
3. **Reorders and re-parses**: Arguments are reordered to put the action first, then the command re-invokes itself using ArgumentParser's native subcommand parsing
4. **Maintains identical behavior**: Both argument orders produce exactly the same result

### Technical Implementation

- Uses ArgumentParser's `@Argument(parsing: .captureForPassthrough)` as a fallback
- Dynamically extracts valid actions from existing subcommands using `Self.configuration.subcommands`
- Includes safety checks for edge cases (empty arrays, index bounds, infinite recursion)

### How to test locally

1. **Test standard subcommand usage** (should work as before):
   ```bash
   tuist xcodebuild build -workspace MyApp.xcworkspace -scheme MyApp
   tuist xcodebuild test -workspace MyApp.xcworkspace -scheme MyAppTests
   ```

2. **Test flexible argument ordering** (new functionality):
   ```bash
   tuist xcodebuild -workspace MyApp.xcworkspace build -scheme MyApp
   tuist xcodebuild -scheme MyApp -workspace MyApp.xcworkspace test
   tuist xcodebuild -derivedDataPath /tmp/build -workspace MyApp.xcworkspace build
   ```

3. **Test edge cases**:
   ```bash
   # Action at the end
   tuist xcodebuild -workspace MyApp.xcworkspace -scheme MyApp build
   
   # Mixed parameters
   tuist xcodebuild -workspace MyApp.xcworkspace -destination 'platform=iOS Simulator,name=iPhone 15' build -scheme MyApp
   
   # Error handling
   tuist xcodebuild -workspace MyApp.xcworkspace invalidaction  # Should show helpful error
   tuist xcodebuild  # Should show help message
   ```

4. **Verify help and tab completion still work**:
   ```bash
   tuist xcodebuild --help
   tuist xcodebuild build --help
   ```

All tests should pass and both argument orders should produce identical results.

### CODE REVIEW

File: cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift [hunk 1]
```diff
@@ -33,7 +33,44 @@ public struct XcodeBuildCommand: AsyncParsableCommand, TrackableParsableCommand,
...
+
+        let validActions = Self.configuration.subcommands.compactMap { $0.configuration.commandName }
+        
+        guard let actionIndex = arguments.firstIndex(where: { validActions.contains($0) }) else {
+            throw ValidationError("No valid action found. Valid actions are: \(validActions.joined(separator: ", "))")
+        }
+        
+        guard actionIndex < arguments.count else {
+            throw ValidationError("Invalid action index.")
+        }
```
Comment (pepicrft) [inline cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift]: In which scenario would the `actionIndex` be higher than `arguments.count`?

**Review by pepicrft — APPROVED** [review]

File: cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift (hunk 1 above)
Comment (yusufozgul) [reply cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift]: It's currently impossible. But I thought it might be possible with future developments. Therefore, throwing an error instead of a crash seemed better.

Comment (pepicrft) [reply cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift]: Let's drop it. If the scenario happens is a bug, and I'd avoid programming defensively against future bugs.

**Comment by pepicrft** [comment]
@fortmarek thoughts on this one?

File: cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift (hunk 1 above)
Comment (yusufozgul) [reply cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommand.swift]: Thanks @pepicrft, removed this guard check

**Comment by pepicrft** [comment]
Thanks for addressing the comment @yusufozgul. Let's add the acceptance test to ensure we don't introduce a regression and we should be good to merge once CI is green.

**Comment by yusufozgul** [comment]
Thanks for review @fortmarek, I made some improvements for command handling and added acceptance tests for all subcommands and command correction.

Can you review again please 🙏  @fortmarek @pepicrft

**Comment by yusufozgul** [comment]
done @pepicrft thanks a lot

**Comment by pepicrft** [comment]
@yusufozgul just rebased from `main` and hoisted the reorderer command to the root in the aim of fixin the compilation error, but it still doesn't go through. The Swift compiler fails to compile the `run` function of the new command. Could you take a look?

```
---
4.	While evaluating request ASTLoweringRequest(Lowering AST to SIL for file "/Users/pepicrft/src/github.com/tuist/tuist/cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommandReorderer.swift")
5.	While silgen emitFunction SIL function "@$s8TuistKit26XcodeBuildCommandReordererV3runyyYaKF".
 for 'run()' (at /Users/pepicrft/src/github.com/tuist/tuist/cli/Sources/TuistKit/Commands/XcodeBuild/XcodeBuildCommandReorderer.swift:19:5)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x000000010916ae24 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x0000000109168c5c llvm::sys::RunSignalHandlers() + 112
2  swift-frontend           0x000000010916b460 SignalHandler(int) + 360
3  libsystem_platform.dylib 0x00000001825c6744 _sigtramp + 56
4  swift-frontend           0x0000000103b3dccc (anonymous namespace)::Transform::transform(swift::Lowering::ManagedValue, swift::Lowering::AbstractionPattern, swift::CanType, swift::Lowering::AbstractionPattern, swift::CanType, swift::SILType, swift::Lowering::SGFContext) + 6888
5  swift-frontend           0x0000000103a8e588 swift::Lowering::Conversion::emit(swift::Lowering::SILGenFunction&, swift::SILLocation, swift::Lowering::ManagedValue, swift::Lowering::SGFContext) const + 520
6  swift-frontend           0x0000000103a8f1a8 swift::Lowering::ConvertingInitialization::copyOrInitValueInto(swift::Lowering::SILGenFunction&, swift::SILLocation, swift::Lowering::ManagedValue, bool) + 104
7  swift-frontend           0x0000000103a162c0 void copyOrInitValuesInto<((anonymous namespace)::ImplodeKind)1>(swift::Lowering::Initialization*, llvm::ArrayRef<swift::Lowering::ManagedValue>&, swift::CanType, swift::SILLocation, swift::Lowering::SILGenFunction&) + 692
8  swift-frontend           0x0000000103a0915c swift::Lowering::ArgumentSource::getAsSingleValue(swift::Lowering::SILGenFunction&, swift::Lowering::SGFContext) && + 388
9  swift-frontend           0x0000000103a8dfe4 swift::Lowering::SILGenFunction::emitConvertedRValue(swift::SILLocation, swift::Lowering::Conversion const&, swift::Lowering::SGFContext, llvm::function_ref<swift::Lowering::ManagedValue (swift::Lowering::SILGenFunction&, swift::SILLocation, swift::Lowering::SGFContext)>) + 300
10 swift-frontend           0x0000000103a09678 swift::Lowering::ArgumentSource::getConverted(swift::Lowering::SILGenFunction&, swift::Lowering::Conversion const&, swift::Lowering::SGFContext) && + 124
11 swift-frontend           0x0000000103a49eb4 (anonymous namespace)::ArgEmitter::emit(swift::Lowering::ArgumentSource&&, swift::Lowering::AbstractionPattern, std::__1::optional<swift::AnyFunctionType::Param>) + 5940
12 swift-frontend           0x0000000103a37064 (anonymous namespace)::ArgEmitter::emitSingleArg(swift::Lowering::ArgumentSource&&, swift::Lowering::AbstractionPattern, std::__1::optional<swift::AnyFunctionType::Param>) + 204
13 swift-frontend           0x0000000103a48484 (anonymous namespace)::ArgEmitter::emitPreparedArgs(swift::Lowering::PreparedArguments&&, swift::Lowering::AbstractionPattern) + 252
14 swift-frontend           0x0000000103a54678 (anonymous namespace)::CallSite::emit(swift::Lowering::SILGenFunction&, swift::Lowering::AbstractionPattern, swift::CanTypeWrapper<swift::SILFunctionType>, (anonymous namespace)::ParamLowering&, llvm::SmallVectorImpl<swift::Lowering::ManagedValue>&, llvm::SmallVectorImpl<(anonymous namespace)::DelayedArgument>&, swift::ForeignInfo const&) && + 660
15 swift-frontend           0x0000000103a53fc0 (anonymous namespace)::CallEmission::emitArgumentsForNormalApply(swift::Lowering::AbstractionPattern, swift::CanTypeWrapper<swift::SILFunctionType>, swift::ForeignInfo const&, llvm::SmallVectorImpl<swift::Lowering::ManagedValue>&, std::__1::optional<swift::SILLocation>&) + 800
16 swift-frontend           0x0000000103a3bb68 (anonymous namespace)::CallEmission::apply(swift::Lowering::SGFContext) + 1072
17 swift-frontend           0x0000000103a43214 swift::Lowering::SILGenFunction::emitGetAccessor(swift::SILLocation, swift::SILDeclRef, swift::SubstitutionMap, swift::Lowering::ArgumentSource&&, bool, bool, swift::Lowering::PreparedArguments&&, swift::Lowering::SGFContext, bool, std::__1::optional<swift::ActorIsolation>) + 820
```

**Comment by yusufozgul** [comment]
@pepicrft Interesting crash happened when accessing commandName using keyPath. I used long format to prevent lint from converting it to keyPath.

**Comment by yusufozgul** [comment]
@pepicrft can we merge this?

**Comment by pepicrft** [comment]
> @pepicrft can we merge this?

I just fixed a linting issue (unrelated to your changes). Let's wait until CI passes and then merge.

**Comment by fortmarek** [comment]
The other tests passed previously, so we should be good to merge without waiting for the couple of pipelines still in progress.
